examples/seismic/benchmark.py

```diff
@@ -109,17 +111,32 @@ def run(problem, **kwargs):
     """
     A single run with a specific set of performance parameters.
     """
-    run = tti_run if problem == 'tti' else acoustic_run
+    setup = tti_setup if problem == 'tti' else acoustic_setup
+    options = {}
+
     time_order = kwargs.pop('time_order')[0]
     space_order = kwargs.pop('space_order')[0]
-    run(space_order=space_order, time_order=time_order, **kwargs)
+    autotune = kwargs.pop('autotune')
+
+    # Should a specific block-shape be used? Useful if one wants to skip
+    # the autotuning pass as a good block-shape is already known
+    block_shape = as_tuple(kwargs.pop('block_shape'))
+    if all(block_shape):
+        if autotune:
+            warning("Skipping autotuning (using explicit block-shape `%s`)"
+                    % str(block_shape))
+            autotune = False
+        # This is quite hacky, but it does the trick
+        for d, bs in zip(['x', 'y', 'z'], block_shape):
```

Comment:
Ouch

Reply:
this is horrible, but I'd need to break too many abstractions to do it cleanly...

Reply:
Yeah but this won't even run in 2D

Reply:
in fact, it will. Unfortunately, the entire benchmark.py won't, probably, crashing at argument parsing time

Reply:
That's even scarier :D